src/components/UI/Inputs/RadioInput.tsx

```diff
@@ -73,6 +73,9 @@ interface IRadioInputProps
 
   /* Customization props for the root label element. */
   rootProps?: Omit<ComponentPropsWithRef<'label'>, 'htmlFor'>;
+
+  /* className for applying a class to the root. */
+  className?: string;
```

Comment:
This is not really how this was designed to work. You can pass props to the root component by using `rootProps`. The rest of the props are meant to be passed to the internal `input` component. That's why the props type inherits from the base `input` component.

Keeping it like this would be pretty confusing. Imagine setting the `name` prop, which affects the input, but setting the `className` prop affects another component.

Reply:
Yeah I see your point, The reason why I've added the `className` prop here to allow style overrides with `styled`. See https://github.com/giantswarm/happa/pull/1550/files#diff-654d39fbe9c6c7c4e51037c47c39a319R38-R40

Reply:
I understand why you did it, but it shouldn't be necessary. Why doesn't this work without adding all that?

```javascript
<RadioInput rootProps={{ className }} />
```

Reply:
This should be to your linking: 0c8ab74

Reply:
damn... we need to switch to [this](https://github.com/emotion-js/emotion/tree/master/packages/babel-preset-css-prop) asap

Reply:
This worked for me:

```javascript
import { ClassNames, css } from '@emotion/core';

const SomeCss = css`
  display: none;
`;

<ClassNames>
  {({ css: css2 }) => (
    <RadioInput
      value='naah'
      id='some-radio'
      name='inputtrickery'
      label='Some text'
      rootProps={{ className: css2(SomeCss) }}
    />
  )}
</ClassNames>
```

Reply:
So do you're fine with the solution in 0c8ab74?

Reply:
I'm fine either way, it's up to you which one of the 2 solutions fits our codebase best 😄